### ISSUE TITLE
myLocation does not return in reasonable time

### ISSUE THREAD

**Issue by vargab95** [liodali/osm_flutter#482]
I implemented the following simple app to play around with osm_flutter. However, my attempt has failed. My idea was to open the map, get the user's location, go to that location and zoom in. I tested it with an android emulator and on a real phone as well and in both cases, I see only the "calling mylocation" message and it does not print the position even after minutes. Maybe I'm doing something wrong. The version is 0.70.3.

```dart
import 'package:flutter/material.dart';
import 'package:flutter_osm_plugin/flutter_osm_plugin.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: MyMapPage(),
    );
  }
}

class MyMapPage extends StatefulWidget {
  const MyMapPage({super.key});

  @override
  State<MyMapPage> createState() => _MyMapPageState();
}

class _MyMapPageState extends State<MyMapPage> {
  late MapController _controller;

  @override
  void initState() {
    super.initState();
    _controller = MapController.withUserPosition();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: OSMFlutter(
        controller: _controller,
        osmOption: const OSMOption(),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () {
          print('calling mylocation');
          _controller.myLocation().then((position) {
            print(position.toString());
          });
        },
        child: const Icon(Icons.location_searching),
      ),
    );
  }
}
```

**Comment by YoavSl** [comment]
+1

**Comment by KseniaKeren** [comment]
+1

**Comment by davidnir1** [comment]
+1, really need this.. thanks!

**Comment by TomerIluz** [comment]
+2!

**Comment by liodali** [comment]
we published new version contain fix thnx for @vargab95  for the fix
the new version `0.70.4`
in future versions we will support external userLocationHandler where you can manage it from your app directly or use any plugin/package for that

**Comment by liodali** [comment]
check our latest version : `1.0.0-dev.1`

**Comment by vargab95** [comment]
Thanks for the fix! It works fine with 1.0.0-dev.3. The above example prints "calling mylocation" closely followed by the location data as it was expected.
